### ISSUE TITLE
Support config option to start without ExUnit hooks.

### ISSUE THREAD

**Issue by jamesotron** [PSPDFKit-labs/bypass#9]
Hi there.

I'm using Bypass with espec. It'd be amazing if it were possible to just call `Bypass.open` without it exploding - I'm currently starting and stopping the bypass supervisor in my test setup and teardown.

**Comment by MSch** [comment]
Hi James,

Bypass needs ExUnit's `on_exit` callbacks to check if a request actually arrived at the started Plug (and to rethrow an exception that happened). If you can come up with a solution that makes Bypass work with ExUnit as well was espec I'd be happy to merge it. I'm assuming adding an application configuration to tell bypass whether it's running in ExUnit or espec would make sense, what do you think?

**Comment by jamesotron** [comment]
Okay. I'll have a think about how we could have it work. I'd suggest that we abstract it into separate "drivers" for each test framework.

**Comment by MSch** [comment]
Hi @jamesotron, just wanted to check if you found a workaround in the meantime?

**Comment by mwean** [comment]
@MSch this is what I did that seems to work against master:

```elixir
  before_all do
    Application.ensure_all_started(:bypass)
  end

  let :bypass, do: start_bypass

  finally do
    case Bypass.Instance.call(bypass.pid, :on_exit) do
      :ok ->
        :ok
      :ok_call ->
        :ok
      {:error, :not_called} ->
        raise ESpec.AssertionError, "No HTTP request arrived at Bypass"
      {:error, :unexpected_request} ->
        raise ESpec.AssertionError, "Bypass got an HTTP request but wasn't expecting one"
      {:exit, {class, reason, stacktrace}} ->
        :erlang.raise(class, reason, stacktrace)
    end
  end

  def start_bypass do
    case Supervisor.start_child(Bypass.Supervisor, []) do
      {:ok, pid} ->
        port = Bypass.Instance.call(pid, :port)
        %Bypass{pid: pid, port: port}
      other ->
        other
    end
  end
```

**Comment by MSch** [comment]
This looks excellent. We are not using espec ourselves, but if you submit a PR on top of https://github.com/PSPDFKit-labs/bypass/pull/36/ we can merge this in. 👍

**Comment by MSch** [comment]
FYI I just released 0.8.0 of bypass with @tompave's ESpec support included ❤️
